src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +381,31 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
```

Comment:
RuntimeException like IllegalStateException or NullPointerException serves the same goal as assertions - they provide a way to detect a logic error. Detecting logic error earlier in a development cycle or even in a customer deployment (without the requirement to give customers special debug build) IMO worth few extra line of code.